web_app/db/crud.py

```diff
@@ -326,3 +326,40 @@ def open_position(self, position_id: uuid) -> Position | None:
             position.status = Status.OPENED.value
             self.write_to_db(position)
         return position.status
+    
+    def get_unique_users(self) -> list[str]:
+        """
+        Fetches all unique users from the Position table.
+        :return: list of unique user ids
+        """
+        with self.Session() as db :
+            try:
+                # Query for distinct user IDs from the Position table
+                unique_users = db.query(Position.user_id).distinct().all()
+                return [user[0] for user in unique_users]  # Extract user_id from tuple
+            except SQLAlchemyError as e:
+                logger.error(f"Error fetching unique users: {str(e)}")
+                return []
+            
+    def get_all_amounts_for_opened_positions(self) -> dict[str, float]:
+        """
+        Fetches the total amount for all users where position status is 'OPENED'.
+        
+        :return: A dictionary with user IDs as keys and the total amount as values.
+        """
+        with self.Session() as db:
+            try:
+                # Query all positions with status OPENED and group by user_id
+                opened_positions = (
+                    db.query(Position.user_id, db.func.sum(Position.amount).label('total_amount'))
+                    .filter(Position.status == Status.OPENED.value)
+                    .group_by(Position.user_id)
```

Comment:
Why do you need to group by `users`?
```
total_opened_amount = (
    db.query(db.func.sum(Position.amount))
    .filter(Position.status == Status.OPENED.value)
    .scalar()
)
```

Reply:
here i'm returning a dic of user ids & their corresponding amount for opened positions.  I'll rename the variables for better clarity
or am I to return something different from this?

Reply:
Yeah, we need to have result as a amount of all `amount` for `opened` position
Just copy/paste what I made as an example above